### PR TITLE
Feature(thirdPartyUser):add third party user

### PR DESCRIPTION
**Pull request by 15168440402**

### CODE REVIEW

File: src/Contracts/Masa.Auth.Contracts.Admin/Infrastructure/Constants/CacheKey.cs [hunk 1]
```diff
@@ -13,6 +13,9 @@ public sealed class CacheKey
     const string ACCOUNT_LOGIN_PRE = "account_login:";
     const string LDAP_OPTIONS_PRE = "ldap_options:";
     const string USER_SYSTEM_DATA_PRE = "user_system_data:";
+    const string UPDATE_USER_PHONENUMBER = "update_user_phoneNumber:";
```
Comment (MayueCif) [inline src/Contracts/Masa.Auth.Contracts.Admin/Infrastructure/Constants/CacheKey.cs:16]: _code

File: src/Services/Masa.Auth.Service.Admin/Application/Subjects/Commands/UpdateStaffBasicInfoCommandValidator.cs [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+// Licensed under the Apache License. See LICENSE.txt in the project root for license information.
+
+namespace Masa.Auth.Service.Admin.Application.Subjects.Commands;
+
+public class UpdateStaffBasicInfoCommandValidator : AbstractValidator<UpdateStaffBasicInfoCommand>
+{
+    public UpdateStaffBasicInfoCommandValidator()
+    {
+        RuleFor(command => command.Staff.DisplayName).Required().MaxLength(50);
+        RuleFor(command => command.Staff.PhoneNumber).Required().Phone();
```
Comment (MayueCif) [inline src/Services/Masa.Auth.Service.Admin/Application/Subjects/Commands/UpdateStaffBasicInfoCommandValidator.cs:11]: PhoneNumber Email 不是单独更新吗

File: src/Services/Masa.Auth.Service.Admin/Application/Subjects/CommandHandler.cs [hunk 3]
```diff
@@ -171,17 +174,71 @@ public async Task UpsertUserAsync(UpsertUserCommand command)
             user = await _userRepository.FindAsync(u => u.Id == userModel.Id);
             if (user is not null)
             {
-                await VerifyUserRepeatAsync(userModel.Id, userModel.PhoneNumber, default, userModel.Email, userModel.IdCard, default);
-                user.Update(userModel.Name, userModel.DisplayName!, userModel.IdCard, userModel.CompanyName, userModel.Gender);
+                user.Update(userModel.Name, userModel.DisplayName!, userModel.IdCard, userModel.CompanyName, userModel.Department, userModel.Gender);
```
Comment (MayueCif) [inline src/Services/Masa.Auth.Service.Admin/Application/Subjects/CommandHandler.cs:177]: 缺少IdCard 重复校验

File: src/Services/Masa.Auth.Service.Admin/Application/Subjects/CommandHandler.cs [hunk 4]
```diff
@@ -171,17 +174,71 @@ public async Task UpsertUserAsync(UpsertUserCommand command)
...
+            throw new UserFriendlyException("This user data does not exist");
+
+        var verifiyKey = CacheKey.VerifiyUserPhoneNumberResultKey(user.Id.ToString(), user.PhoneNumber);
+        var success = await _cache.GetAsync<bool>(verifiyKey);
+        if(success)
+        {
+            var key = CacheKey.UpdateUserPhoneNumberKey(userDto.Id.ToString(), userDto.PhoneNumber);
+            if (await _sms.VerifyMsgCodeAsync(key, userDto.VerificationCode))
+            {
+                user.UpdateAvatar(userDto.PhoneNumber);
```
Comment (MayueCif) [inline src/Services/Masa.Auth.Service.Admin/Application/Subjects/CommandHandler.cs:236]: ？

Comment (MayueCif) [reply src/Services/Masa.Auth.Service.Admin/Application/Subjects/CommandHandler.cs:236]: 缺少PhoneNumber 重复校验

**Review by MayueCif — CHANGES_REQUESTED** [review]
@15168440402

File: src/Services/Masa.Auth.Service.Admin/Application/Subjects/Commands/UpdateStaffBasicInfoCommandValidator.cs (hunk 2 above)
Comment (15168440402) [reply src/Services/Masa.Auth.Service.Admin/Application/Subjects/Commands/UpdateStaffBasicInfoCommandValidator.cs:11]: Staff do not need a separate interface to modify mobile phone numbers and mailboxes

**Review by MayueCif — APPROVED** [review]
LGTM

**Pull request merged**